geonode/maps/qgis_server_views.py

```diff
@@ -343,15 +343,13 @@ def get_context_data(self, **kwargs):
             config = snapshot_config(snapshot, map_obj, request.user,
                                      access_token)
         # list all required layers
-        layers = Layer.objects.all()
         map_layers = MapLayer.objects.filter(
             map_id=mapid).order_by('stack_order')
         context = {
             'config': json.dumps(config),
             'create': False,
-            'layers': layers,
             'resource': map_obj,
-            'map_layers': map_layers,
+            'layers': map_layers,
```

Comment:
changed map_layers into layers, so it gets the correct variable. I'm not sure if you made a typo here or not, please check @boney-bun

Reply:
this is a function related to embedding an iframe.
i've tried your PR on the testing server. 
however, it loads the whole layers instead.
this is an example:
![screen shot 2017-12-05 at 1 06 04 am](https://user-images.githubusercontent.com/4602383/33568310-a1414670-d958-11e7-88bc-7577d4fb7dd7.png)

so, i think we still need to figure out this parts.

Reply:
This fix right here is to avoid just that.